src/plot_api/plot_api.js

```diff
@@ -1686,6 +1686,74 @@ function _restyle(gd, aobj, traces) {
     };
 }
 
+/**
+ * Converts deprecated layout attributes to the current API
+ * to ensure backwards compatibility.
+ *
+ * @param layoutObj
+ */
+function cleanDeprecatedLayoutAttributes(layoutObj) {
+
+    // Support old-style update of the title's font
+    var isTitlefontSet = layoutObj.titlefont;
+    var isFontInTitleSet = layoutObj.title && layoutObj.title.font;
+    var isFontInTitleNotSet = !isFontInTitleSet;
+    if(isTitlefontSet && isFontInTitleNotSet) {
+
+        // Use string attribute because initiating a new title object
+        // to be able to specify a font property on it would require to
+        // know the potentially existing `title.text` property.
+        layoutObj['title.font'] = layoutObj.titlefont;
+        delete layoutObj.titlefont;
+    }
+
+    // Note, that updating x-axis and y-axis title fonts
+    // was never supported unless (i) using string
+    // attributes or (ii) passing `xaxis.title` / `yaxis.title`
+    // again. And these cases are covered by
+    // helpers.cleanLayout anyways.
```

Comment:
Hmm. Could you share an example of `(ii)`?

Reply:
I think the gist should be that if `xaxis` or `yaxis` is passed to relayout, a `replot` will get triggered. And `replot` calls and thus cleans deprecated attributes anyways. 
However, I've decided to throw away the code of `cleanDeprecatedLayoutAttributes` because `cleanDeprecatedLayoutAttributeStrings` effectively does the same thing to `titlefont`. Though I renamed `cleanDeprecatedLayoutAttributeStrings` now. Hope the code gets a little bit clearer this way.

Reply:
Nicely done :ok_hand: